components/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/Utils.java

```diff
@@ -69,6 +74,58 @@ public static Path getCarbonHome() {
     }
 
     /**
+     * Returns the Runtime Home directory path. If {@code runtime.home} system property is not found, gets the
+     * {@code RUNTIME_HOME_ENV} system property value and sets to the runtime home.
+     *
+     * @return the Runtime Home directory path
+     */
+    public static Path getRuntimeHome() {
+        String runtimeHome = System.getProperty(Constants.RUNTIME_HOME);
+        if (runtimeHome == null) {
+            runtimeHome = System.getenv(Constants.RUNTIME_HOME_ENV);
+            System.setProperty(Constants.RUNTIME_HOME, runtimeHome);
+        }
+        return Paths.get(runtimeHome);
+    }
+
+    /**
+     * get the current Runtime name.
+     *
+     * @return the Runtime name
+     */
+    public static String getRuntime() {
+        return System.getProperty(Constants.RUNTIME);
+    }
+
+    /**
+     * Get the runtime home path for a runtime.
+     *
+     * @return Path to runtime home of given runtime.
+     * @throws IOException if there error while reading runtime root.
+     */
+    public static Path getRuntimeHome(String runtimeName) throws IOException {
```

Comment:
In the case of CarbonHome which is the Server Home patch, but in the case of Runtime, there is the runtime name and runtime patch.